### PR TITLE
Fix: code quality and docs

### PR DESCRIPTION
**Pull request by ayaankhan98**
#### Description of Change

- fixed code quality 
- added docs
- added test cases

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: sorting/insertion_sort.cpp [hunk 1]
```diff
@@ -1,36 +1,112 @@
...
+    tests();
+
+    /// For user insteraction
+    int n;
+    std::cout << "Enter the length of your array : ";
+    std::cin >> n;
+    int *arr = new int[n];
+    std::cout << "Enter any " << n << " Numbers for Unsorted Array : ";
+
+    for (int i = 0; i < n; i++) std::cin >> arr[i];
```
Comment (Panquesito7) [inline sorting/insertion_sort.cpp:102]: Please make it look like C++ code:

```suggestion
    for (int i = 0; i < n; i++) { std::cin >> arr[i]; }
```

File: sorting/insertion_sort.cpp [hunk 2]
```diff
@@ -1,36 +1,112 @@
...
+
+    insertionSort(arr, n);
 
-    // Output
     std::cout << "\nSorted Array : ";
-    for (int i = 0; i < n; i++) {
-        std::cout << Array[i] << "\t";
-    }
+    for (int i = 0; i < n; i++) std::cout << arr[i] << " ";
```
Comment (Panquesito7) [inline sorting/insertion_sort.cpp:107]: Same as above.

File: sorting/insertion_sort.cpp [hunk 3]
```diff
@@ -1,36 +1,116 @@
...
+ * (Insertion Sort)](https://en.wikipedia.org/wiki/Insertion_sort)
+ *
+ * \author
+ *
+ * \details
+ * Insertion sort is a simple sorting algorithm that builds the final
+ * sorted array one at a time. It is much less efficient compared to
+ * other sorting algorithms like heap sort, merge sort or quick sort.
+ * However it has several advantages such as
+ * <pre>
```
Comment (kvedala) [inline sorting/insertion_sort.cpp:14]: Pre tag not needed.

File: sorting/insertion_sort.cpp [hunk 4]
```diff
@@ -1,36 +1,116 @@
...
+ * 1 - easy to implement
```
Comment (kvedala) [inline sorting/insertion_sort.cpp:15]:
```suggestion
 * 1. Easy to ....
```

File: sorting/insertion_sort.cpp [hunk 5]
```diff
@@ -1,36 +1,116 @@
...
+ * 2 - For small set of data it is quite efficient
+ * 3 - More efficient that other Quadratic complexity algorithms like
+ *     Selection sort or bubble sort.
+ * 4 - It's stable that is it does not change the relative order of
+ *     elements with equal keys
+ * 5 - Works on hand means it can sort the array or list as it receives.
+ * </pre>
```
Comment (kvedala) [inline sorting/insertion_sort.cpp:22]: To be removed

File: sorting/insertion_sort.cpp [hunk 6]
```diff
@@ -1,36 +1,116 @@
...
+ *
+ * It is based on the same idea that people use to sort the playing cards in
+ * their hands.
+ * the algorithms goes in the manner that we start iterating over the array
+ * of elements as soon as we find a unsorted element that is a misplaced
+ * element we place it at a sorted position.
+ *
+ * Suppose initially we have
+ * <pre>
```
Comment (kvedala) [inline sorting/insertion_sort.cpp:31]: Missing close tag

File: sorting/insertion_sort.cpp [hunk 7]
```diff
@@ -1,36 +1,116 @@
...
+ * we place it at the correct sorted position thus the array in this iteration
+ * becomes
+ *
+ * 2 3 4 5 1
+ *
+ * we does not do anything with 5 and move on to the next iteration and select
+ * 1 which is misplaced and place it at correct position. Thus, we have
+ *
+ * 1 2 3 4 5
+ * </pre>
```
Comment (kvedala) [inline sorting/insertion_sort.cpp:50]: The whole block will render poorly

**Comment by kvedala** [comment]
For future reference, give 48 hrs for some other admin to review as well.

**Comment by ayaankhan98** [comment]
> For future reference, give 48 hrs for some other admin to review as well.

oh my bad luck i was confused between two PR's i was supposed to merge that other one but i merged this one. and that other one is still unmerged, sorry for this, well i had opened another PR in regard with this issue please review that one #931.

**Comment by kvedala** [comment]
> > For future reference, give 48 hrs for some other admin to review as well.
> 
> oh my bad luck i was confused between two PR's i was supposed to merge that other one but i merged this one. and that other one is still unmerged, sorry for this, well i had opened another PR in regard with this issue please review that one #931.

ok. please be cautious next time :)

**Pull request merged**